pkg/osquery/runtime/osqueryinstance.go

```diff
@@ -259,6 +278,66 @@ func (i *OsqueryInstance) Exited() <-chan struct{} {
 	return i.errgroup.Exited()
 }
 
+// ReloadKatcExtension can be called on a running osquery instance to reload its KATC extension
+// manager server, to add new KATC tables or update existing KATC tables' configurations
+// without restarting the entire instance.
+func (i *OsqueryInstance) ReloadKatcExtension(ctx context.Context) error {
+	ctx, span := traces.StartSpan(ctx)
+	defer span.End()
+
+	if !i.instanceStarted() {
+		return errors.New("instance not started, cannot reload extension")
+	}
+
+	i.emsLock.Lock()
+	if katcServer, ok := i.extensionManagerServers[katcExtensionName]; ok {
+		// KATC extension manager server already exists -- we must stop it so that we can start a new one.
+		// We created the KATC extension manager server by calling StartOsqueryExtensionManagerServer with
+		// allowRestart=true so that we can shut it down here without triggering a full shutdown of the
+		// errgroup.
+		katcServer.Shutdown(ctx)
+		delete(i.extensionManagerServers, katcExtensionName)
+		i.slogger.Log(ctx, slog.LevelInfo,
+			"shut down KATC extension manager server in preparation for reload",
+		)
+	}
+	i.emsLock.Unlock()
+
+	if err := i.startKatcExtensionManagerServer(ctx); err != nil {
+		return fmt.Errorf("starting katc server: %w", err)
+	}
+
+	return nil
+}
+
+func (i *OsqueryInstance) instanceStarted() bool {
```

Comment:
maybe worth adding a comment here that we expect callers to manage emsLock while using

Reply:
Ah I'll have this one lock too, seems simpler -- will update